api/core/model_runtime/model_providers/tongyi/llm/llm.py

```diff
@@ -512,6 +515,19 @@ def _convert_prompt_messages_to_tongyi_messages(self, prompt_messages: list[Prom
 
         return tongyi_messages
 
+    def object_to_dict(self, obj):
```

Comment:
This `obj` should be a `ToolCall` instance. Here is its structure: [link](https://github.com/langgenius/dify/blob/5a99aeb864092d2e3b31e5547ce827ab290e4447/api/core/model_runtime/entities/message_entities.py#L111).

It's better to add a type hint and avoid using the magic methods like `__dict__`.

Reply:
ToolCall has parameter ToolCallFunction, it alse need to transfer to dict.So I think my way is better.

Reply:
I referred to the call of alibaba qwen‘s way

Reply:
How about using the `model_dump` method provided by Pydantic?

Reply:
Fine,I have tested just now. It is the best way with **model_dump**